metl/model_encoder.py

```
@@ -0,0 +1,121 @@
+        Strict is off by default when loading from a checkpoint file, and on when loading models from Zenodo.
+        """
+        try:
+            ppdb = PandasPdb().read_pdb(pdb_file)
+        except Exception as e:
+            raise Exception(f"{str(e)} \n\n PDB file could not be read by PandasPDB. It may be incorrectly formatted.")
+
+        groups = ppdb.df['ATOM'].groupby('residue_number')
+        wt_seq = []
+        for group_name, group_data in groups:
+            wt_seq.append(seq1(group_data.iloc[0]['residue_name']))
+        wildtype = ''.join(wt_seq)
+
+        if self.strict:
+            err_str = "Strict mode is on because a METL model that we trained was used. Wildtype and PDB sequences must match."
+            err_str += " To ignore the sequence mismatch, pass strict=False to the load function you used."
+            assert wildtype == wt, err_str
+
+    def validate_variants(self, variants, wt):
+        """
+        Variants much be validated only after conversion to 0 based!
+        """
+        wt_len = len(wt)
+        for index, variant in enumerate(variants):
+            split = variant.split(',')
+            for mutation in split:
+                from_amino_acid = mutation[0]
+                to_amino_acid = mutation[-1]
+                location = int(mutation[1:-1])
+
+                errors = []
+
+                if location <= 0 or location >= wt_len-1:
+                    error_str = f"The position for the mutation is {location} but it needs to be between 0 "
+                    error_str += f"and {len(wt)-1} if 0-based and 1 and {len(wt)} if 1-based."
+                    errors.append(error_str)
+                elif wt[location] != from_amino_acid:
+                    errors.append(f"Wildtype at position {location} is {wt[location]} but variant had {from_amino_acid}. Check the variant input.")
+
+                if len(errors) != 0:
```

When do you want to raise the exception? It looks like this will raise the exception as soon as you encounter the first bad mutation, in which case I"m not sure why it is necessary to build up a list of error messages.

---

I think it was me not realizing initially that this would've broken with an out of bounds (how it was last commit) and seeing the possibility of both errors could be thrown. I can just have it be an error string here.